src/google-calendar.ts

```diff
@@ -296,3 +304,82 @@ export function checkForAutoCreateMeetingFlag(): boolean {
 
   return false;
 }
+
+function addButtonToGmailCal(quickAddDialog: HTMLElement) {
+  // skip if our button is already added
+  if (document.querySelector("#jitsi_button_quick_add")) {
+    return;
+  }
+  const tabEvent = quickAddDialog.querySelector(".IFQP9d.n1rxgc.SFQtGf");
+  if (tabEvent) {
+    const tabPanel = document.createElement("content");
+    tabPanel.setAttribute("role", "tabpanel");
+    tabPanel.setAttribute("id", "jitsi_button_quick_add_content");
+
+    tabPanel.innerHTML = `
```

Comment:
we prefer to avoid innerHTML calls because there is a very high XSS risk if this is ever modified to take user input as part of the string. you can either:

1. modify this to use `document.createElement` / `document.<element>.appendChild` for each subelement
2. use DOMPurify to sanitize the string, preferably with a TrustedHTML object so we can enable Trusted Types on this extension: https://github.com/cure53/DOMPurify#what-about-dompurify-and-trusted-types

Reply:
re: trusted types, this doesn't necessarily have to be done as part of this PR but here's an example: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/require-trusted-types-for

to set the content security policy in an extension, use https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/#content-security-policy

Reply:
Looks like this code is not something new and I have just taken it from the previous approved [PR](https://github.com/brave/brave-talk-gcalendar-extension/commit/2262aea0eb70848b7d442c23b69feafc8ab95bd7#:~:text=content%22)%3B-,tabPanel,-.setAttribute() (see lines 43-61). This code has also passed the prior [security review](https://github.com/brave/security/issues/690) as well.

Reply:
@diracdeltas - i seem to recall a conversation with @tackley and the two of us regarding that code, and that there wasn't a means to fix it. So, we let it go. Does that sound right? Thanks!

Reply:
can we ask someone to join this thread and explain exactly how to address the issue. i don't know how to fix it.

Reply:
@diracdeltas  Thanks for the feedback. I'll be updating the `innerHTML` with `createElement` and `append` function.

Reply:
great, thx for doing this